ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactAnalyzer.java

```diff
@@ -67,6 +73,17 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
     }
 
     AlterTableCompactDesc desc = new AlterTableCompactDesc(tableName, partitionSpec, type, isBlocking, mapProp);
+    Table table = getTable(tableName);
+    WriteEntity.WriteType writeType = null;
+    if (AcidUtils.isTransactionalTable(table)) {
+      setAcidDdlDesc(desc);
+      writeType = WriteType.DDL_EXCLUSIVE;
```

Comment:
cloud you please explain a little bit why we choose DDL_EXCLUSIVE for transactional table?  does it works the same for insert only tables?